Tests fail with GCC 12 (maybe change in error strings for regex library or similar?)

Hi!

All hlwl (including0.9.5) do not build (actually fail the test) with gcc 12

```
__________________________ test_foreach_invalid_flag ___________________________

hlwm = <conftest.HlwmBridge object at 0x7fce3fad9270>

    def test_foreach_invalid_flag(hlwm):
        hlwm.call_xfail('foreach C --filter-typo=X tags. echo C') \
            .expect_stderr('"" has no child named "--filter-typo=X"')
        hlwm.call_xfail('foreach C --filter-name=( tags. echo C') \
>           .expect_stderr('Parenthesis is not closed')

../tests/test_meta_commands.py:848: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self2 = CompletedProcess(args=['/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/herbstclient', '-n', 'forea..., stdout='', stderr='foreach: Cannot parse flag "--filter-name=(": Mismatched \'(\' and \')\' in regular expression\n')
needle = 'Parenthesis is not closed', regex = True

    def f(self2, needle, regex=True):
        if regex:
>           assert re.search(needle, self2.stderr)
E           assert None
E            +  where None = <function search at 0x7fce47771d80>('Parenthesis is not closed', 'foreach: Cannot parse flag "--filter-name=(": Mismatched \'(\' and \')\' in regular expression\n')
E            +    where <function search at 0x7fce47771d80> = re.search
E            +    and   'foreach: Cannot parse flag "--filter-name=(": Mismatched \'(\' and \')\' in regular expression\n' = CompletedProcess(args=['/<<PKGBUILDDIR>>/obj-x86_64-linux-gnu/herbstclient', '-n', 'forea..., stdout='', stderr='foreach: Cannot parse flag "--filter-name=(": Mismatched \'(\' and \')\' in regular expression\n').stderr

../tests/conftest.py:107: AssertionError
```


Context:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1012943
Full Buildlog (amd64)
https://buildd.debian.org/status/fetch.php?pkg=herbstluftwm&arch=amd64&ver=0.9.5-1&stamp=1659993132&raw=0

--- a/tests/test_meta_commands.py
+++ b/tests/test_meta_commands.py
@@ -845,7 +845,7 @@ def test_foreach_invalid_flag(hlwm):
     hlwm.call_xfail('foreach C --filter-typo=X tags. echo C') \
         .expect_stderr('"" has no child named "--filter-typo=X"')
     hlwm.call_xfail('foreach C --filter-name=( tags. echo C') \
-        .expect_stderr('Parenthesis is not closed')
+        .expect_stderr(r'(Parenthesis is not closed|Mismatched.*\(.*\).*in regular)')
     hlwm.call_xfail('foreach C tags. get_attr --filter-name=X') \
         .expect_stderr('has no attribute "--filter-name=X"')
 

